projects/fhi-angular-highcharts/src/lib/services/diagram-type-group.service.ts

```diff
@@ -171,123 +176,144 @@ export class DiagramTypeGroupService {
+    }
+    if (diagramType.id === DiagramTypes.columnAndLine.id) {
+      if (this.notTwoUnits(diagramType)) return true;
+      if (this.notTwoUnitsInSeries(diagramType)) return true;
+      if (this.notAllUnitsFoundInSeries(diagramType)) return true;
+    }
 
-      case DiagramTypeIdValues.column:
-        return this.disableColumn();
+    return false;
+  }
 
-      case DiagramTypeIdValues.columnAndLine:
-        return this.disableColumnAndLine();
+  private isBarOrColumnType(diagramType: DiagramType): boolean {
+    return (
+      diagramType.id === DiagramTypes.bar.id ||
+      diagramType.id === DiagramTypes.barStacked.id ||
+      diagramType.id === DiagramTypes.column.id ||
+      diagramType.id === DiagramTypes.columnAndLine.id ||
+      diagramType.id === DiagramTypes.columnStacked.id
+    );
+  }
 
-      case DiagramTypeIdValues.columnStacked:
-        return this.disableColumnStacked();
+  private isMapType(diagramType: DiagramType): boolean {
+    return (
+      diagramType.id === DiagramTypes.mapFylker.id ||
+      diagramType.id === DiagramTypes.mapFylker2019.id ||
+      diagramType.id === DiagramTypes.mapFylker2023.id
+    );
+  }
 
-      case DiagramTypeIdValues.mapFylker:
-      case DiagramTypeIdValues.mapFylker2019:
-      case DiagramTypeIdValues.mapFylker2023:
-        return this.disableMap();
+  private isMapOrPieType(diagramType: DiagramType): boolean {
+    return this.isMapType(diagramType) || diagramType === DiagramTypes.pie;
+  }
 
-      case DiagramTypeIdValues.pie:
-        return this.disablePie();
+  private isAnyTypeButTable(diagramType: DiagramType): boolean {
+    return diagramType.id !== DiagramTypes.table.id;
+  }
 
-      default:
-        return false;
-    }
+  private isAnyTypeButTableOrColumnAndLine(diagramType: DiagramType): boolean {
+    return !(
+      diagramType.id === DiagramTypes.table.id || diagramType.id === DiagramTypes.columnAndLine.id
+    );
   }
 
-  private disableBar(): boolean {
+  private hasFlaggedData(diagramType: DiagramType): boolean {
     if (this.series.length > 1 && this.flaggedSeries?.length !== 0) {
-      this.diagramTypeDisabledWarnings.bar = this.diagramTypeDisabledWarningsText.flaggedData;
+      this.updateDisabledWarnings(diagramType.id, msgId.hasFlaggedData);
       return true;
     }
     return false;
   }
 
-  private disableBarStacked(): boolean {
-    if (this.disableBar()) {
-      this.diagramTypeDisabledWarnings.barStacked =
-        this.diagramTypeDisabledWarningsText.flaggedData;
+  private moreThanOneSeries(diagramType: DiagramType): boolean {
+    if (this.series.length > 1) {
+      this.updateDisabledWarnings(diagramType.id, msgId.moreThanOneSeries);
       return true;
     }
     return false;
   }
 
-  private disableColumn(): boolean {
-    if (this.disableBar()) {
-      this.diagramTypeDisabledWarnings.column = this.diagramTypeDisabledWarningsText.flaggedData;
+  private notAllUnitsFoundInSeries(diagramType: DiagramType): boolean {
```

Comment:
`allUnitsFoundInSeries` is hard to read. Simplify to  `const allUnitsFoundInSeries = this.diagramOptions.units.every((unit) =>
      this.series.some((serie) => serie.unitId === unit.id)
    );`?

Reply:
Nice! Much better 🙂